### PR TITLE
Rm old expansion

### PR DESCRIPTION
**Pull request by rwst**
It removes superfluous expansion code and moves declarations/definitions of global `series` and `series_invfunc` to `series.h/cpp`.

### CODE REVIEW

File: symengine/series_piranha.cpp [hunk 1]
```diff
@@ -155,6 +201,35 @@ Expression UPSeriesPiranha::convert(const Number &x) {
...
+RCP<const Basic> UPSeriesPiranha::as_basic() const {
+    RCP<const Symbol> x = symbol(var_);
+    umap_basic_num dict_;
+    for (const auto& it : p_) {
+        if (it.first != 0) {
+            const RCP<const Basic> &term = SymEngine::mul(SymEngine::pow(x, SymEngine::integer(it. second.degree())), it.first.get_basic());
+            RCP<const Number> coef;
+            RCP<const Basic> t;
+            Add::as_coef_term(term, outArg(coef), outArg(t));
+            Add::dict_add_term(dict_, coef, t);
```
Comment (isuruf) [inline symengine/series_piranha.cpp]: `term` can be an `Add` for `x**0`. `dict_add_term` doesn't handle that. `coef_dict_add_term` does

**Comment by isuruf** [comment]
`series` and `series_invfunc` can return `RCP<const SeriesCoeffInterface>` (after changing SeriesCoeffInterface to inherit from `Number`). If a user needs a `umap_short_basic` then the user can call `as_dict`. 

Advantage of this method is that `series` and `series_invfunc` can return any type inheriting `SeriesCoeffInterface`.

**Comment by rwst** [comment]
Yeah I already thought this was not satisfying.

**Comment by rwst** [comment]
Ok I'm ready with this. I'll create a `test_series.cpp` when I deal with Flint series. Please review.

**Comment by isuruf** [comment]
+1. Looks good to me.
Btw, is there any change in the speed of series benchmarks with the new expansion? I wanted to check this myself, but didn't have the time to do it.

**Comment by rwst** [comment]
Thanks. I seem to remember a slight slowdown (original `expansion.cpp` --> `URatP`), which will be only relevant later if you don't use Flint (why would anyone?).

**Comment by certik** [comment]
Piranha now supports series expansion? Is it using your code, or did @bluescarni implement something on his own?

**Comment by rwst** [comment]
@certik We still use Piranha polys to make our own expansion code. What @isuruf referred to was his abstraction layer, on top of which everything now sits.

**Pull request merged**